include/seqan3/alignment/pairwise_alignment_edit_distance_unbanded.hpp

```diff
@@ -0,0 +1,548 @@
+            // It is used for the mask computation and setting the initial score (the minus one is there because of the Ukkonen trick).
+            unsigned localMaxErrors = std::min<unsigned>(max_errors, query.size() - 1);
+            score_mask = (word_type)1 << (localMaxErrors % word_size);
+            last_block = std::min(localMaxErrors / word_size, block_count - 1);
+            _score = localMaxErrors + 1;
+        }
+
+        word_type vp0{~static_cast<word_type>(0)};
+        word_type vn0{0};
+
+        vp.resize(block_count, vp0);
+        vn.resize(block_count, vn0);
+        bit_masks.resize((alphabet_size + 1) * block_count, 0);
+
+        // encoding the letters as bit-vectors
+        for (unsigned j = 0; j < query.size(); j++)
+        {
+            unsigned i = block_count * to_rank(query[j]) + j / word_size;
+            bit_masks[i] |= (word_type)1 << (j % word_size);
+        }
+
+        add_state();
+    }
+    //!\}
+
+private:
+
+    //!\brief One compute step in one column.
+    template <bool with_overflow_check>
+    inline void compute_step(word_type b, word_type & hp, word_type & hn, word_type & vp, word_type & vn, word_type & carry_d0, word_type & carry_hp, word_type & carry_hn)
+    {
+        word_type x, d0, t;
+
+        x = b | vn;
+        t = vp + (x & vp) + (with_overflow_check ? carry_d0 : 0);
+
+        d0 = (t ^ vp) | x;
+        hn = vp & d0;
+        hp = vn | ~(vp | d0);
+
+        if constexpr(with_overflow_check)
+            carry_d0 = (carry_d0 != (word_type)0) ? t <= vp : t < vp;
+
+        x = (hp << 1) | (with_overflow_check ? carry_hp : hp0);
+        vn = x & d0;
+        vp = (hn << 1) | ~(x | d0) | (with_overflow_check ? carry_hn : 0);
+
+        if constexpr(with_overflow_check)
+        {
+            carry_hp = hp >> (word_size - 1);
+            carry_hn = hn >> (word_size - 1);
+        }
+    }
+
+    //!\brief Increase or decrease the score.
+    inline void advance_score(word_type P, word_type N, word_type mask)
+    {
+        if ((P & mask) != (word_type)0)
+            _score++;
+        else if ((N & mask) != (word_type)0)
+            _score--;
+    }
+
+    //!\brief Decrement the last active cell position.
+    inline bool prev_last_active_cell()
+    {
+        score_mask >>= 1;
+        if (score_mask != (word_type)0)
+            return true;
+
+        last_block--;
+        if (is_global && last_block == (unsigned)-1)
+            return false;
+
+        score_mask = (word_type)1 << (word_size - 1);
+        return true;
+    }
+
+    //!\brief Increment the last active cell position.
+    inline void next_last_active_cell()
+    {
+        score_mask <<= 1;
+        if (score_mask)
+            return;
+
+        score_mask = 1;
+        last_block++;
+    }
+
+    //!\brief Use the ukkonen trick and update the last active cell.
+    inline bool update_last_active_cell()
+    {
+        // updating the last active cell
+        while (!(_score <= max_errors))
```

Comment:
`>` ?

Reply:
I find this easier to read. Because, you know that after the while loop `_score <= max_errors` is true.